libs/mibao-ui/src/lib/tab/tab.tsx

```diff
@@ -1,23 +1,138 @@
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState
+} from 'react'
 import {
   Tabs as ChakraTabs,
   Tab as ChakraTab,
-  TabList,
+  TabList as ChakraTabList,
+  TabsProps as ChakraTabsProps,
+  TabProps as ChakraTabProps,
   TabPanels,
   TabPanel,
-  TabsProps
+  useTab,
+  Box
 } from '@chakra-ui/react'
+import styles from './tab.module.scss'
+import { ThemeTypings } from '@chakra-ui/styled-system'
+import { omit } from '../../utils'
 
-export { TabList, TabPanel, TabPanels, TabsProps }
+export { TabPanel, TabPanels }
+
+export type TabVariant = 'line' | 'unstyled' | 'enclosed' | 'enclosed-colored' | 'soft-rounded' | 'solid-rounded'
+export type TabsAlign = 'start' | 'end' | 'center' | 'space-between'
+type ColorSchemes = ThemeTypings['colorSchemes'] | string
+
+export interface TabsProps extends Omit<ChakraTabsProps, 'variant' | 'orientation' | 'align'> {
+  variant?: TabVariant
+  align?: TabsAlign
+}
+
+const TABS_CONTEXT_VALUE = {
+  index: 0,
+  variant: 'line' as TabVariant,
+  colorScheme: 'primary' as ColorSchemes,
+  align: 'start' as TabsAlign
+}
+const TabsContext = createContext(TABS_CONTEXT_VALUE)
+
+const TabActiveLine: React.FC<{
+  colorScheme: ColorSchemes
+  offset: number
+  width: number
+}> = ({
+  colorScheme,
+  offset,
+  width
+}) => (
+  <Box
+    w="1px"
+    bg={ colorScheme === 'black' ? 'black' : `${colorScheme}.600` }
+    h="2px"
+    position="absolute"
+    bottom="-2px"
+    left={0}
+    transformOrigin="left"
+    transition={'0.2s'}
+    transform={`translateX(${offset}px) scaleX(${width})`}
+  />
+)
 
 export const Tabs: React.FC<TabsProps> = (props) => {
+  const [index, setIndex] = useState(props.defaultIndex ?? props.index ?? 0)
+  const variant = props.variant ?? 'line'
+  const colorScheme = props.colorScheme ?? 'primary'
+  const align = props.align ?? 'start'
+
+  const onChange = useCallback((index) => {
+    props?.onChange?.(index)
+    setIndex(index)
+  }, [props])
+
   return (
-    <ChakraTabs {...props} colorScheme="primary">
-      {props.children}
+    <ChakraTabs
+      {...omit(props, ['align'])}
+      aligh={align === 'space-between' ? undefined : align}
+      colorScheme={colorScheme}
+      variant={variant}
+      onChange={onChange}
+    >
+      <TabsContext.Provider value={{ index, variant, colorScheme, align }}>
+        {props.children}
+      </TabsContext.Provider>
     </ChakraTabs>
   )
 }
 
-export const Tab: React.FC = (props) => {
-  return <ChakraTab {...props}>{props.children}</ChakraTab>
+export const TabList: React.FC = ({ children }) => {
+  const { index, variant, colorScheme, align } = useContext(TabsContext)
+  const tabListRef = useRef<HTMLDivElement>(null)
+  const [width, setWidth] = useState(0)
+  const [offset, setOffset] = useState(0)
+  useEffect(() => {
+    if (tabListRef.current && variant === 'line') {
+      const elements = Array.from<HTMLButtonElement>(tabListRef.current.querySelectorAll(`.${styles.tab}`))
```

Comment:
If `styles.tab` is just used in selector, class name of a plain string is enough, no need to introduce it from `css file`